### PR TITLE
feat(http): Add type to MockBackend.connections

### PR DESCRIPTION
**Pull request by lacolaco**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

`MockBackend#connections` and `MockBackend#pendingConnections` are typed as `any`. It's not editor-friendly.

**What is the new behavior?**

They'll be typed correctly. 

**Does this PR introduce a breaking change?** (check one with "x")
```
[x] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

Some wrong typed code may break compilation. No impacts on correct `MockBackend` uses.

**Other information**:

### CODE REVIEW

File: modules/@angular/http/testing/mock_backend.ts [hunk 1]
```diff
@@ -174,7 +177,7 @@ export class MockBackend implements ConnectionBackend {
    *
    * This property only exists in the mock implementation, not in real Backends.
    */
-  connections: any;  //<MockConnection>
+  get connections(): Observable<MockConnection> { return this._connections; };
```
Comment (vicb) [inline modules/@angular/http/testing/mock_backend.ts]: why not `connections: Observable<MockConnection>` ?

File: modules/@angular/http/testing/mock_backend.ts [hunk 2]
```diff
@@ -191,13 +195,13 @@ export class MockBackend implements ConnectionBackend {
    *
    * This property only exists in the mock implementation, not in real Backends.
    */
-  pendingConnections: any;  // Subject<MockConnection>
```
Comment (vicb) [inline modules/@angular/http/testing/mock_backend.ts]: same here

File: modules/@angular/http/testing/mock_backend.ts (hunk 1 above)
Comment (laco0416) [inline modules/@angular/http/testing/mock_backend.ts]: `connections` should be writable in `MockBackend`, but I think it shouldn't be exposed as it is. 
From outside of MockBackend, I think it should be read-only.

File: modules/@angular/http/testing/mock_backend.ts (hunk 2 above)
Comment (laco0416) [inline modules/@angular/http/testing/mock_backend.ts]: ditto

File: modules/@angular/http/testing/mock_backend.ts (hunk 1 above)
Comment (laco0416) [inline modules/@angular/http/testing/mock_backend.ts]: fixed

File: modules/@angular/http/testing/mock_backend.ts (hunk 2 above)
Comment (laco0416) [inline modules/@angular/http/testing/mock_backend.ts]: fixed

**Comment by laco0416** [comment]
@vicb I refactored it. could you review again?

**Comment by laco0416** [comment]
@vicb what status is this PR? can I do any actions?

**Comment by vicb** [comment]
Need to review again and evaluates when we can merge (due to the breaking change)

**Comment by laco0416** [comment]
@vicb At now, could I ask you review this again?

**Comment by vicb** [comment]
I'll take a look today, sorry for the delay

File: modules/@angular/http/testing/mock_backend.ts [hunk 3]
```diff
@@ -230,7 +231,8 @@ export class MockBackend implements ConnectionBackend {
    *
    * This property only exists in the mock implementation, not in real Backends.
    */
-  connections: any;  //<MockConnection>
+  connections: Observable<MockConnection>;
```
Comment (vicb) [inline modules/@angular/http/testing/mock_backend.ts]: could we make it 
`connections: Observable<MockConnection> = new Subject<MockConnection>;

**Comment by vicb** [comment]
Please make the commit message `feat(html): ...` - the PR does not "fix" a bug and it's easier to spot what should go in 4.x vs 2.x. Thanks.

**Comment by laco0416** [comment]
@vicb I see. I'll do that. 
BTW, I'm thinking of the type of `MockBackend#connections`. probably it's should not be `Observable` but `EventEmitter` as its documentation.

```
   * {@link EventEmitter}
   * of {@link MockConnection} instances that have been created by this backend. Can be subscribed
   * to in order to respond to connections.
   *
```

So I'll rewrite my code and ask review again.

**Comment by laco0416** [comment]
@vicb updated. Is the failure of Travis my fault?

File: modules/@angular/http/testing/mock_backend.ts [hunk 4]
```diff
@@ -230,7 +228,7 @@ export class MockBackend implements ConnectionBackend {
    *
    * This property only exists in the mock implementation, not in real Backends.
    */
-  connections: any;  //<MockConnection>
+  connections: EventEmitter<MockConnection> = new EventEmitter<MockConnection>();
```
Comment (vicb) [inline modules/@angular/http/testing/mock_backend.ts]: no need to type the var here (inferred from the new, same below)

**Comment by vicb** [comment]
The error was probably a flake, I have restarted the CI build

File: modules/@angular/http/testing/mock_backend.ts (hunk 4 above)
Comment (laco0416) [inline modules/@angular/http/testing/mock_backend.ts]: fixed.

**Comment by laco0416** [comment]
@vicb could you review again?

**Comment by vicb** [comment]
Closing as outdated

**Pull request closed without merging**